js/renovation/ui/scheduler/workspaces/base/virtual-row.tsx

```diff
@@ -0,0 +1,55 @@
+import {
+  Component, ComponentBindings, JSXComponent, OneWay,
+} from 'devextreme-generator/component_declaration/common';
+import { addHeightToStyle } from '../utils';
+import { RowProps, Row } from './row';
+import { VirtualCell } from './virtual-cell';
+
+export const viewFunction = (viewModel: VirtualRow): JSX.Element => (
+  <Row
+    // eslint-disable-next-line react/jsx-props-no-spreading
+    {...viewModel.restAttributes}
+    style={viewModel.style}
+    className={viewModel.classes}
+  >
+    {viewModel.virtualCells.map((_, index) => (
+      <VirtualCell key={index.toString()} />
+    ))}
+  </Row>
+);
+
+@ComponentBindings()
+export class VirtualRowProps extends RowProps {
+  @OneWay() height?: number;
+
+  @OneWay() cellsCount = 1;
+}
+
+@Component({
+  defaultOptionRules: null,
+  view: viewFunction,
+})
+export class VirtualRow extends JSXComponent(VirtualRowProps) {
+  get style(): { [key: string]: string | number | undefined } {
+    const { height } = this.props;
+    const { style } = this.restAttributes;
+
+    return addHeightToStyle(height, style);
+  }
+
+  get classes(): string {
+    const { className } = this.props;
+
+    return `dx-scheduler-virtual-row ${className}`;
```

Comment:
This variable has been declared in `scheduler/workspaces/ui.scheduler.work_space.js`. Can we use it here too?
> const VIRTUAL_ROW_CLASS = 'dx-scheduler-virtual-row';

Reply:
Usually, we do not use constants from old code (as far as I know, no one does in renovation). An we were explicitly asked **not** to use constants for class names in the renovated code, so I would not do that, at least for now.